### PR TITLE
Fix up nint code after latest changes to use compiler nint.

### PR DESCRIPTION
**Pull request by eerhardt**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs [hunk 1]
```diff
@@ -115,11 +115,7 @@ private Utf8Span(ReadOnlySpan<byte> rawData)
             // Allow retrieving references to just past the end of the span (but shouldn't dereference this).
 
             Debug.Assert(index <= (uint)Length, "Caller should've performed bounds checking.");
-#if SYSTEM_PRIVATE_CORELIB
             return ref Unsafe.AddByteOffset(ref DangerousGetMutableReference(), index);
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: There are no UIntPtr overloads in the public `System.Runtime.CompilerServices.Unsafe`. I expect that you will need to keep the cast to make it compile.

We may look at adding the UIntPtr overloads for `Unsafe.Add/Subtract` in `System.Runtime.CompilerServices.Unsafe`. Now that we have the nuint type, they make a lot more sense than before.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: Do you think I should log an "api proposal" for the UIntPtr overloads?

Comment (jkotas) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: Sounds good to me.

Comment (jkotas) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: I assume that it will be no-brainer to get approved.

Comment (jkotas) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: cc @GrabYourPitchforks

Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs:118]: I see now why my local build succeeded. I built Utf8String.Experimental using the default TFM (which is `net5.0`). This code isn't compiled during the `net5.0` build.

**Review by GrabYourPitchforks — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
I agree we should add overloads to existing methods where appropriate. The `Unsafe` class has several good candidates. `Marshal.AllocHGlobal` and friends would also work.

There were some earlier comments (https://github.com/dotnet/runtime/pull/36159#issuecomment-626124709, https://github.com/dotnet/runtime/pull/36159#issuecomment-626186223) about changing the parameters on existing methods from `IntPtr` to `nint`. But on reflection adding new overloads might be better. `Marshal.AllocHGlobal(nuint)` is a more sensible overload than `Marshal.AllocHGlobal(nint)`, for instance, and it would allow you to allocate more than 2GB in a 32-bit process assuming the process is large address aware.

File: src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs (hunk 1 above)
Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs]: Logged https://github.com/dotnet/runtime/issues/36232

**Pull request merged**